easypy/logging.py

```diff
@@ -126,6 +127,52 @@ def get_console_handler():
             return handler
 
 
+class ThreadControl(logging.Filter):
+    """
+    Used by ContextLoggerMixin .solo and .suppressed methods to control logging to console
+    To use, add it to the logging configuration as a filter in the console handler
+
+        ...
+        'filters': {
+            'thread_control': {
+                '()': 'easypy.logging.ThreadControl'
+            }
+        },
+        'handlers': {
+            'console': {
+                'class': 'logging.StreamHandler',
+                'filters': ['thread_control'],
+            },
+
+    """
+
+    CONTEXT = ThreadContexts(counters='silenced')
+
+    # we use this ordered-dict to track which thread is currently 'solo-ed'
+    # we populate it with some initial values to make the 'filter' method
+    # implementation more convenient
+    SELECTED = OrderedDict({None: None})
+    IDX_GEN = count()
+    LOCK = threading.RLock()
+
+    @classmethod
+    @contextmanager
+    def solo(cls):
+        try:
+            with cls.LOCK:
+                idx = next(cls.IDX_GEN)
+                cls.SELECTED[idx] = threading.current_thread()
+            yield
+        finally:
+            cls.SELECTED.pop(idx)
+
+    def filter(self, record):
+        selected = self.SELECTED[self.SELECTED._OrderedDict__root.prev.key]  # hacky way for quickly getting to the last element
```

Comment:
This will not work in newer versions of Python (3.5+ I think). `next(reversed(self.SELECTED))` will get you the last key and is also _O(1)_ even in 3.4.

Reply:
you don't say...